pulley/src/interp.rs

```diff
@@ -2612,25 +2612,47 @@ impl ExtendedOpVisitor for Interpreter<'_> {
 
     fn bitcast_int_from_float_32(&mut self, dst: XReg, src: FReg) -> ControlFlow<Done> {
         let val = self.state[src].get_f32();
-        self.state[dst].set_u32(val.to_bits());
+        let bits = if val.is_nan() {
+            0x7fc00000
+        } else {
+            val.to_bits()
+        };
```

Comment:
I'm surprised this is necessary: the intent of `canonicalize-nan.wast` is to test Cranelift's NaN-canonicalization pass, and this pass operates on the machine-independent IR (CLIF) so it should work for compilation to Pulley as well as any other target. The Pulley virtual machine shouldn't have to unconditionally canonicalize NaNs.

Could you check whether the pass is producing NaN-canonicalization sequences (which look like compares to test for NaN, then a select operator)?

I'm wondering whether this might have something to do with the `has_vector_support` boolean in `Context::canonicalize_nans()`; the comment there (that riscv64 is the only architecture without vector support) is no longer accurate now that we have Pulley in-tree; I'm not sure but perhaps that's related?

Reply:
Oh I think this might be my fault, in the [contribution docs for pulley](https://github.com/bytecodealliance/wasmtime/blob/main/pulley/CONTRIBUTING.md) it recommends running the test suite directly via `wasmtime wast .../*.wast`. When `tests/misc_testsuite/simd/canonicalize-nan.wast` is executed that's actually expected to fail (it even fails on native) and it requires an extra `-W nan-canonicalization` flag. 

Chris is right though in that the manual canonicalization here shouldn't be necessary. @Lordworms you'll find though that `-W nan-canonicalization` isn't fully supported just yet:

```
Error: failed to run script file './tests/misc_testsuite/simd/canonicalize-nan.wast'

Caused by:
    0: failed directive on ./tests/misc_testsuite/simd/canonicalize-nan.wast:7:1
    1: Compilation error: Unsupported feature: should be implemented in ISLE: inst = `v10 = fcmp.f32x4 uno v7, v7`, type = `Some(types::I32X4)`
```

That being said this PR still gets the `simd_f32x4_rounding.wast` test passing, so with the canonicalization removed it should still be good to flag that test as passing

Reply:
I am new to this project and didn't find it. Sorry for that.